packages/api/src/router/post.ts

```diff
@@ -6,4 +8,47 @@ export const postRouter = createTRPCRouter({
       orderBy: { createdAt: "desc" },
     });
   }),
+  createPost: protectedProcedure
+    .input(
+      z.object({
+        title: z.string().min(1).max(300),
+        contents: z.string().min(1),
+      }),
+    )
+    .mutation(async ({ ctx, input }) => {
+      await ctx.db.post.create({
+        data: {
+          title: input.title,
+          contents: input.contents,
+        },
+      });
+    }),
+  updatePostByID: protectedProcedure
+    .input(
+      z.object({
+        id: z.number().min(0),
+        title: z.string().optional(),
+        contents: z.string().optional(),
+      }),
+    )
+    .mutation(async ({ ctx, input }) => {
+      await ctx.db.post.update({
+        where: {
+          id: input.id,
+        },
+        data: {
+          title: input.title,
+          contents: input.contents,
+        },
+      });
+    }),
+  deletePostByID: protectedProcedure
+    .input(z.object({ id: z.number().min(0) }))
```

Comment:
you can also use `z.number().nonnegative()`, might make more sense as a reader

Reply:
just fixed it!